src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import { Button, Card, Col, Form, Row } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import { MovieCard } from "../movie-card/movie-card";
+
+export const ProfileView = ({ movies }) => {
+  const localUser = JSON.parse(localStorage.getItem("user"));
+  const storedToken = localStorage.getItem("token");
+
+  if (!localUser) {
+    console.log("User not found in localStorage");
+    return;
+  }
+
+  if (!storedToken) {
+    console.log("Token not found in localStorage");
+    return;
+  }
+
+  const [name, setName] = useState(localUser?.Name || "");
+  const [email, setEmail] = useState(localUser?.Email || "");
+  const [birthday, setBirthday] = useState(localUser?.Birthday || "");
+
+  const fav = movies.filter((movie) => {
+    return localUser.FavoriteMovies.includes(movie._id);
+  });
+
+  const handleToggleFavorite = (updatedUserDetails) => {
+    localStorage.setItem("user", JSON.stringify(updatedUserDetails));
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const data = {
+      Name: name,
+      Email: email,
+      Birthday: birthday,
+    };
+
+    fetch(
+      `https://radiant-river-68463-0f7c4a72bc48.herokuapp.com/users/${localUser._id}`,
+      {
+        method: "PUT",
+        body: JSON.stringify(data),
+        headers: {
+          Authorization: `Bearer ${storedToken}`,
+          "Content-Type": "application/json",
+        },
+      }
+    ).then(async (response) => {
+      if (response.ok) {
+        alert("Profile details updated successful");
+
+        const updatedUser = await response.json();
+        localStorage.setItem("user", JSON.stringify(updatedUser));
+
+        setName(updatedUser.Name);
+        setEmail(updatedUser.Email);
+        setBirthday(updatedUser.Birthday);
+
+        window.location.reload();
+      } else {
+        alert("Profile details updation failed");
+      }
+    });
+  };
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <Form.Group controlId="formName">
+        <Form.Label>Username:</Form.Label>
+        <Form.Control
+          type="text"
+          value={name}
+          onChange={(e) => setName(e.target.value)}
+          required
+          minLength="4"
+        />
+      </Form.Group>
+      <Form.Group controlId="formEmail">
+        <Form.Label>Email:</Form.Label>
+        <Form.Control
+          type="email"
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+          required
+        />
+      </Form.Group>
+      <Form.Group controlId="formBdate">
+        <Form.Label>Birthday:</Form.Label>
+        <Form.Control
+          type="date"
+          value={birthday}
```

Comment:
Here you're setting the value of the birthday as it is from the API. You need to transform it to the format suggested in the warning message.
However, It's a dangerous practice to set the value of inputs with the current values. User placeholders instead. As for displaying user info, you can add a separate container for it.